packages/neos-ui-editors/src/Editors/Button/index.js

```diff
@@ -0,0 +1,157 @@
+    dataSourceIdentifier: props.options.dataSourceIdentifier,
+    dataSourceUri: props.options.dataSourceUri,
+    dataSourceAdditionalData: props.options.dataSourceAdditionalData,
+    dataSourceDisableCaching: Boolean(props.options.dataSourceDisableCaching)
+});
+
+@neos(globalRegistry => ({
+    i18nRegistry: globalRegistry.get('i18n'),
+    dataSourcesDataLoader: globalRegistry.get('dataLoaders').get('DataSources')
+}))
+@connect($transform({
+    focusedNodePath: selectors.CR.Nodes.focusedNodePathSelector
+}))
+export default class ButtonEditor extends Component {
+    static propTypes = {
+        commit: PropTypes.func.isRequired,
+        value: PropTypes.any,
+        className: PropTypes.string,
+        options: PropTypes.shape({
+            allowEmpty: PropTypes.bool,
+            multiple: PropTypes.bool,
+            disabled: PropTypes.bool,
+            values: PropTypes.objectOf(PropTypes.shape({
+                label: PropTypes.string,
+                icon: PropTypes.string,
+                iconActive: PropTypes.string,
+                disabled: PropTypes.bool
+            })),
+
+            dataSourceIdentifier: PropTypes.string,
+            dataSourceUri: PropTypes.string,
+            dataSourceDisableCaching: PropTypes.bool,
+            dataSourceAdditionalData: PropTypes.objectOf(PropTypes.any)
+        }).isRequired,
+
+        i18nRegistry: PropTypes.object.isRequired
+    };
+
+    static defaultOptions = {
+        allowEmpty: false,
+        multiple: false,
+        disabled: false
+    };
+
+    state = {
+        active: [],
+        isLoading: false,
+        buttons: []
+    };
+
+    initialValueType = 'string';
+
+    constructor(props) {
+        super(props);
+
+        const {commit, options, value} = props;
+        this.initialValueType = Array.isArray(value) ? 'array' : 'string';
+        if (options.multiple && this.initialValueType === 'string') {
+            console.warn(`Misconfiguration in property "${props.identifier}". Multiple is activated but value type seems to be "string" but should be "array".`);
+        }
+
+        this.state = {
+            active: Array.isArray(value) ? value : (value ? [value] : []),
+            buttons: this.hasDataSource() ? [] : this.flattenValues(options.values)
+        };
+        this.commit = commit;
+    }
+
+    hasDataSource() {
+        return this.props.options.dataSourceIdentifier || this.props.options.dataSourceUri;
+    }
+
+    componentDidMount() {
+        if (this.hasDataSource()) {
+            this.loadOptions();
+        }
+    }
+
+    loadOptions() {
+        this.setState({isLoading: true});
+        this.props.dataSourcesDataLoader.resolveValue(getDataLoaderOptionsForProps(this.props), this.props.value)
+            .then(buttons => {
+                this.setState({
+                    isLoading: false,
+                    buttons
+                });
+            });
+    }
+
+    flattenValues(values) {
```

Comment:
```suggestion
    createButtonsFromConfiguration(configuredValues) {
```
For clarity:
This function does more than just flatten a hashmap of values to an array. The function name should reflect that.